src/sentry/testutils/asserts.py

```diff
@@ -45,6 +50,22 @@ def assert_status_code(response, minimum: int, maximum: int | None = None):
     )
 
 
+def verify_project_deletion(org: Organization, platform: str):
+    # Poll the database to check if the project was deleted
+    # the timeout is set to 10 seconds
+    start_time = time.time()
+    while time.time() - start_time < 10:
+        try:
+            # we need to check for the status here because we do soft deletions
+            Project.objects.get(organization=org, slug=platform, status=ObjectStatus.ACTIVE)
+        except ObjectDoesNotExist:
+            # If the project doesn't exist anymore, it's deleted
+            return  # The project was successfully deleted
```

Comment:
```suggestion
        assert not Project.objects.filter(organization=org, slug=platform, status=ObjectStatus.ACTIVE).exists()
```
small optimization (exceptions are more expensive than just boolean check) and it is imo easier to read